src/main/webapp/mapScript.js

```diff
@@ -132,32 +90,90 @@ function addNewTableItem(name, id) {
-    if (markersArray[i].id == contentId) {
-      markersArray[i].marker.setMap(null);
-      markersArray.splice(i, 1);
-      console.log("remove marker from the array and the map");
-      break;
-    }
-  }
+  markersArray[contentId].marker.setMap(null);
+  markersArray.splice(contentId, 1);
+  console.log("remove marker from the array and the map");
+}
 
-  let options = {
-    method: "POST",
+function showSettings(contentId) {
+  var settings = document.getElementsByClassName("marker-setting")[0];
+  document.getElementById("submit-button").onclick = function () {
+    updateMarkerSettings(contentId);
   };
 
-  let URL = "/delete_marker?contentId=" + contentId;
-  await fetch(URL, options)
-    .then((response) => response.json())
-    .then((deletionResult) => {
-      if (deletionResult) {
-        console.log("Successful deletion");
-      } else {
-        console.log("Unsuccessful deletion");
-      }
-    });
+  // Show popup.
+  // setting.style.visibility = "visible";
+  settings.classList.toggle("show");
+  document.getElementById("marker-name").value =
+    markersArray[contentId].data.markerName;
+  document.getElementById("visit-hour").value =
+    markersArray[contentId].data.visitHour;
+  document.getElementById("visit-minute").value =
+    markersArray[contentId].data.visitMinute;
+  document.getElementById("leave-hour").value =
+    markersArray[contentId].data.leaveHour;
+  document.getElementById("leave-minute").value =
+    markersArray[contentId].data.leaveMinute;
+}
+
+function updateMarkerSettings(contentId) {
+  let markerName = document.getElementById("marker-name").value,
+    visitHour = document.getElementById("visit-hour").value,
+    visitMinute = document.getElementById("visit-minute").value,
+    leaveHour = document.getElementById("leave-hour").value,
+    leaveMinute = document.getElementById("leave-minute").value;
+
+  markersArray[contentId].data.markerName = markerName;
+  markersArray[contentId].data.visitHour = visitHour;
+  markersArray[contentId].data.visitMinute = visitMinute;
+  markersArray[contentId].data.leaveHour = leaveHour;
+  markersArray[contentId].data.leaveMinute = leaveMinute;
+
+  let tableItemElemens = document.getElementById("place" + contentId)
+    .childNodes;
+  tableItemElemens[1].innerHTML = markerName;
+}
+
+async function createRoute() {
+  var routeName = document.getElementById("route-name").value;
+  if (routeName == "") {
+    alert("Please add a name to your new route!");
+  } else {
+    var markersData = [];
+    for (var i = 0; i < markersArray.length; i++) {
+      markersData.push(markersArray[i].data);
+    }
+
+    var routeData = {
+      routeName: routeName,
+      markersData: markersData,
+    };
+    console.log(routeData);
+
+    let options = {
+      method: "POST",
+      body: JSON.stringify(routeData),
+      headers: {
+        "Content-Type": "application/json",
+      },
+    };
+
+    await fetch("/storeRoute", options).then();
```

Comment:
It would be good to handle the response from the fetch. At the moment it seems to be ignored. Perhaps create and issue and add a TODO to fix in a future PR.